src/firmware/app/control/BUILD

```diff
@@ -12,17 +12,36 @@ cc_library(
     srcs = ["trajectory_planner.c"],
     hdrs = ["trajectory_planner.h"],
     deps = [
+        "//firmware/app/control:trajectory_planner_impl",
+        "//firmware/shared:physics",
         "//firmware/shared/math:polynomial_1d",
         "//firmware/shared/math:polynomial_2d",
+        "//firmware/shared/math:tbots_math",
         "//firmware/shared/math:vector_2d",
     ],
 )
 
+cc_library(
+    name = "trajectory_planner_impl",
+    srcs = ["trajectory_planner_impl.c"],
+    hdrs = ["trajectory_planner_impl.h"],
+    visibility = [
+        "//visibility:private",
+    ],
+    deps = [
+        "//firmware/shared:physics",
+        "//firmware/shared/math:polynomial_1d",
+        "//firmware/shared/math:polynomial_2d",
+        "//firmware/shared/math:tbots_math",
+    ],
+)
+
 cc_test(
     name = "trajectory_planner_test",
     srcs = ["trajectory_planner_test.cpp"],
     deps = [
         ":trajectory_planner",
+        ":trajectory_planner_impl",
```

Comment:
Do the tests for `trajectory_planner` and `trajectory_planner_impl` depend on one another at all? If not might be a good ideal to break this up into two separate tests so we don't have a huge 2000 line test file. Not _super_ strongly opinionated so non-blocking comment.

Reply:
+1 for breaking the tests up into separate files if possible

Reply:
Broke them up into a test file for impl as well